bundles/persistence/org.openhab.persistence.jdbc/java/org/openhab/persistence/jdbc/internal/JdbcPersistenceService.java

```diff
@@ -0,0 +1,333 @@
+ * # defaults:
+ * #jdbc:sqltype.CALL           =	VARCHAR(200)
+ * #jdbc:sqltype.COLOR          =	VARCHAR(70)
+ * #jdbc:sqltype.CONTACT        =	VARCHAR(6)
+ * #jdbc:sqltype.DATETIME       =	DATETIME
+ * #jdbc:sqltype.DIMMER         =	TINYINT
+ * #jdbc:sqltype.LOCATION       =	VARCHAR(30)
+ * #jdbc:sqltype.NUMBER         =	DOUBLE
+ * #jdbc:sqltype.ROLLERSHUTTER  =	TINYINT
+ * #jdbc:sqltype.STRING         =	VARCHAR(65500)
+ * #jdbc:sqltype.SWITCH         =	VARCHAR(6)
+ * 
+ * # For Itemtype "Number" default decimal digit count (optional, default: 3) 
+ * #jdbc:numberDecimalcount=
+ * 
+ * # T A B L E   O P E R A T I O N S
+ * # Tablename Prefix String (optional, default: "item") 
+ * # for Migration from MYSQL-Bundle set to 'Item'.
+ * #jdbc:tableNamePrefix=Item
+ * 
+ * # Tablename Prefix generation, using Item real names or "item" (optional, default: false -> "item") 
+ * # If set to true, 'tableNamePrefix' is ignored.
+ * #jdbc:tableUseRealItemNames=
+ * jdbc:tableUseRealItemNames=true
+ * 
+ * # Tablename Suffix length (optional, default: 4 -> 0001-9999) 
+ * # For Migration from MYSQL-Bundle set to 0.
+ * #jdbc:tableIdDigitCount=
+ * 
+ * # Rename existing Tables using tableUseRealItemNames and tableIdDigitCount (optional, default: false) 
+ * # USE WITH CARE! Deactivate after Renaming is done!
+ * #jdbc:rebuildTableNames=true
+ * 
+ * # D A T A B A S E  C O N N E C T I O N S
+ * # Some embeded Databases can handle only one Connection (optional, default: configured per database in packet org.openhab.persistence.jdbc.db. * )
+ * # see: https://github.com/brettwooldridge/HikariCP/issues/256
+ * # jdbc.maximumPoolSize = 1
+ * # jdbc.minimumIdle = 1
+ * 
+ * # T I M E K E E P I N G
+ * # (optional, default: false) 
+ * #jdbc:enableLogTime=true
+ * #######################################################################################
+ * 
+ * 
+ * DEVELOPER INFO TO ADD A NEW DATABASE:
+ * 1. Duplicate org.openhab.persistence.jdbc.config.db.JdbcMysql.java with database name Jdbc[database].java
+ * 2. Minimum you have to configure: setDriverClassName("JDBC DRIVERNAME STRING")
+ * 3. Setup OpenHab *.cfg File jdbc:user=, jdbc:password=, jdbc:url=jdbc:[database]:
+ * 4. Run and Test
+ * 5. For adjustments, use default communication-code from JdbcBaseDAO and copy parts to your new Jdbc[databaseName]DAO class an customize it.
+ * 6. Run and Test
+ * 
+ *
+ * @author Helmut Lehmeyer
+ * @since 1.8.0
+ * 
+ * 
+ */
+public class JdbcPersistenceService extends JdbcMapper implements QueryablePersistenceService {
+	static final Logger logger = LoggerFactory.getLogger(JdbcPersistenceService.class);
+
+	protected ItemRegistry itemRegistry;
+
+	/**
+	 * The BundleContext. This is only valid when the bundle is ACTIVE. It is
+	 * set in the activate() method and must not be accessed anymore once the
+	 * deactivate() method was called or before activate() was called.
+	 */
+	@SuppressWarnings("unused")
```

Comment:
Is this SuppressWarnings necessary? The bundleContext is being used in the activate() method...